src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -140,6 +142,50 @@ static constexpr std::chrono::minutes MAX_LEDGER_AGE_ACQUIRE {1};
 // Don't acquire history if write load is too high
 static constexpr int MAX_WRITE_LOAD_ACQUIRE {8192};
 
+// Helper function for LedgerMaster::doAdvance()
+// Returns the minimum ledger sequence in SQL database, if any.
+boost::optional<LedgerIndex>
+minSqlSeq(Application& app)
+{
+    boost::optional<LedgerIndex> seq;
+    auto db = app.getLedgerDB().checkoutDb();
+    *db << "SELECT MIN(LedgerSeq) FROM Ledgers", soci::into(seq);
+    return seq;
+}
+
+// Helper function for LedgerMaster::doAdvance()
+// Return true if candidateLedger should be fetched from the network.
+bool
+shouldAcquire (
+    std::uint32_t const currentLedger,
```

Comment:
Making "by value" function arguments `const` is a bit unusual and I don't think we do this anywhere else.

Reply:
I like const on parameters because I know it's not going to change in the function itself. This function has been around forever like this. The reason it looks new is because as a result of the review process I made it a free function in the cpp file, so I copy & pasted it.